analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.Implementation.cs

```diff
@@ -25,90 +25,71 @@
 
 namespace SonarAnalyzer.Helpers
 {
-    internal abstract partial class KnownType
+    internal sealed partial class KnownType
     {
-        public string TypeName { get; private init; }
+        private readonly IList<string> namespaceParts;
+        private readonly string[] genericParameters;
 
-        internal abstract bool Matches(ITypeSymbol symbol);
+        public string TypeName { get; }
+        public string FullName { get; }
+        public bool IsArray { get; init; }
 
-        internal sealed class RegularKnownType : KnownType
+        public KnownType(string fullName, params string[] genericParameters)
         {
-            private readonly IList<string> namespaceParts;
-            private readonly string[] genericParameters;
+            FullName = fullName;
```

Comment:
_Whitespace nitpick:_ I don't like the empty lines here, they are sort of irregular and doesn't help much.

Can we reorder it to have `var parts` as a first line? Then we could have just a series of initialization. In the same order as the fields & properties are declared above.
```
            var parts = fullName.Split('.');
            namespaceParts = new ArraySegment<string>(parts, 0, parts.Length - 1);
            this.genericParameters = genericParameters;
            TypeName = parts[parts.Length - 1];
            FullName = fullName;
```